capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -116,125 +99,57 @@ public MetricsClient() {}
-            map = new HashMap<>();
-            map.put("week", entity.getProperty("week").toString());
-            map.put("count", Integer.parseInt(entity.getProperty("count").toString()));
-            resultList.add(map);
+            pageView = new WeeklyPageView(Integer.parseInt(entity.getProperty("week").toString()),
+                    year,
+                    Integer.parseInt(entity.getProperty("count").toString()));
+            pageViewsList.add(pageView);
         }
 
-        return resultList;
+        return pageViewsList;
     }
 
     /**
      * Gets all pageViews for the entire system.
-     * @return returns a Map<String, Object> in the following format:
-     *          {
-     *              "firstDate":
-     *                  {
-     *                      "week": <int>
-     *                      "year": <int>
-     *                  },
-     *              "clickData":
-     *                  [
-     *                      {
-     *                          "restaurantName": <String>
-     *                          "week": <int>
-     *                          "year": <int>
-     *                          "numClicks": <int>
-     *                      },
-     *                      ...
-     *                  ]
-     *          }
-     *
-     *          "clickData" represents all the click data that we have.
-     *          "firstDate" represent the first time data was collected in our system.
+     * @return returns List<RestaurantPageViews> with the PageView list inside each instance
+     *         of RestaurantPageViews sorted by week and year in ascending order.
      */
-    public Map<String, Object> getAllPageViews() {
-        Query query = new Query("PageViews");
+    public List<RestaurantPageViews> getAllPageViews() {
+        // Query with double sort so the first element of every restaurant should be
+        // the earliest week and year.
+        Query query = new Query("PageViews")
+                .addSort("week", Query.SortDirection.ASCENDING)
+                .addSort("year", Query.SortDirection.ASCENDING);
         PreparedQuery preparedQuery = dataStore.prepare(query);
         List<Entity> results = preparedQuery.asList(FetchOptions.Builder.withDefaults());
 
-        Map<String, Object> resultMap = new HashMap<>();
-
-        Map<String, Integer> firstPageView = new HashMap<>();
-        // Initialized with max value so any date will change the map at the beginning.
-        firstPageView.put("year", Integer.MAX_VALUE);
-        firstPageView.put("week", Integer.MAX_VALUE);
+        // Map to maintain a reference to each restaurant and make searching for them easy.
+        // The map and the list share the reference to each restaurantPageViews.
+        Map<String, RestaurantPageViews> restaurantNameMap = new HashMap<>();
+        List<RestaurantPageViews> restaurantPageViewsList = new ArrayList<>();
+        RestaurantPageViews currRestaurant;
```

Comment:
Same comment here; this doesn't need to be declared out here.

Reply:
Done. I changed it to winthin the else block because I need the same reference to be both in the map and the list, in order to be able to edit both when the restaurant already exists in the map.